README.md

```diff
@@ -2,31 +2,230 @@
+| Build frontend         | X      | X     |        |      | X     |            |      |       |
+| Build backend          | X      | X     |        |      |       | X          | X    |       |
+| Publishing to pypi.org | X      | X     |        |      |       |            | X    | X     |
+
+DSP-TOOLS uses [poetry](https://python-poetry.org) for all of these tasks. This allows us to use one single tool 
+for all processes, and to keep the number of configuration files at a minimum. 
+
+There are many configuration and metadata files that can be found on the top level of a Python repository. The ones 
+used in the DSP-TOOLS repository are:
+
+| file           | purpose                                                                         |
+|----------------|---------------------------------------------------------------------------------|
+| README.md      | Markdown-formatted infos for developers                                         |
+| pyproject.toml | Modern configuration/metadata file replacing the deprecated files listed below  |
+| .gitignore     | List of files not under version control (won't be uploaded to GitHub)           |
+| .gitmodules    | DSP-TOOLS contains a Git submodule (more infos below)                           |
+| CHANGELOG.md   | Markdown-formatted release notes (must not be edited by hand)                   |
+| LICENSE        | Text file with the license how to use the source code of DSP-TOOLS              |
+| Makefile       | Definition of commands that can be executed with `make [command]`               |
+| poetry.lock    | Pinned versions of all (sub-) dependencies, allows a deterministic installation |
+| mkdocs.yml     | Configuration of `mkdocs`, used to build the documentation webpage              |
+
+In earlier times, there were some more configuration files, but thanks to poetry, they are not necessary anymore:
+
+| Deprecated file      | Replaced by                             | Replaced by                                          |
+|----------------------|-----------------------------------------|------------------------------------------------------|
+| MANIFEST.in          | files to include into distribution      | pyproject.toml: [tool.poetry.include]                |
+| setup.py             | project metadata, dependencies          | pyproject.toml                                       |
+| setup.cfg            | configuration for setuptools            | pyproject.toml                                       |
+| requirements.txt     | all (sub-) dependencies                 | pyproject.toml: [tool.poetry.dependencies]           |
+| dev-requirements.txt | additional dependencies for development | pyproject.toml: [tool.poetry.group.dev.dependencies] |
+| Pipfile              | direct dependencies                     | pyproject.toml: [tool.poetry.dependencies]           |
+| Pipfile.lock         | pinned dependencies                     | poetry.lock                                          |
```

Comment:
I would not document this in the readme. IMHO, the readme should only contain the actual state, not past things that are not relevant anymore.